### PR TITLE
Newsletters: add 102 (2020-06-17)

### PR DESCRIPTION
**Pull request by harding**
- [x] Make suggested changes to #411, get it merged, and rebase this PR on that to allow linking to the *vault* topic @harding 
- [x] Services and client section @bitschmidty 
- [x] `BIPs #910` @bitschmidty

### CODE REVIEW

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      liquidity (the *market makers*), collect their input and output
+      information, and create the transaction template.  This gives the
+      taker knowledge of which inputs fund which outputs for all
+      participants in the coinjoin, but it also ensures that each maker
+      only has knowledge about which of their inputs funds which of
+      their outputs.  The taker directly gains the privacy benefits of
+      the coinjoin and the makers directly gain income for providing
+      liquidity.  If the taker preserves his own privacy, the makers
+      also indirectly gain increased privacy against third party block
+      chain surveilance.  Makers who want guarantees about their privacy
```
Comment (adamjonas) [inline _posts/en/newsletters/2020-06-17-newsletter.md:70]:
```suggestion
      chain surveillance.  Makers who want guarantees about their privacy
```

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  for more details on this proposal.
+
+{% include references.md %}
+{% include linkers/issues.md issues="910" %}
+[coinpool post]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2020-June/017964.html
+[joinpool]: https://gist.github.com/harding/a30864d0315a0cebd7de3732f5bd88f0
+[joinmarket]: https://github.com/JoinMarket-Org/joinmarket-clientserver
+[wasabi]: https://wasabiwallet.io/
+[wabisabi post]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2020-June/017969.html
+[wabisabi paper]: https://github.com/zkSNACKs/WabiSabi/releases/latest/download/WabiSabi.pdf
```
Comment (adamjonas) [inline _posts/en/newsletters/2020-06-17-newsletter.md:166]: I _really_ didn't like that this downloaded on click. Do you think that the link to the mailing list post is enough?

Comment (harding) [reply _posts/en/newsletters/2020-06-17-newsletter.md:166]: Sure, the mailing list post is enough and I'm open to removing the link, but we've linked to PDFs many times before, so I curious to understand what's different about this one?

    $ git grep '\.pdf' -- _posts/en/newsletters/ | wc -l
    49

Comment (adamjonas) [reply _posts/en/newsletters/2020-06-17-newsletter.md:166]: The others look to be hosted online while this downloads to my machine without warning. I don't have a problem with the .pdf format per se.

Comment (harding) [reply _posts/en/newsletters/2020-06-17-newsletter.md:166]: Ah.  Yeah, your browser probably has a built-in preview for PDF files that set their MIME type correctly, but this document has its MIME type set as binary (octet-stream):



Amusingly, the document also is automatically downloaded if you're viewing it in a local-browser-based preview (and this has indeed been the vector for some previous browser security vulnerabilities, though perhaps not as many as non-browser-based PDF vulnerabilities).

In any case, GitHub net setting the correct MIME type is a good enough reason to remove the link.  I did try playing with the `<a>` tag's `type` attribute but, as the HTML5 standard specifies, I'm unable to use this to override the server's MIME type.

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  collaboratively create a transaction template that spends some of
+  their existing UTXOs (inputs) to a new set of UTXOs (outputs).  The
+  way this transaction template is created has implications for the
+  privacy provided by the coinjoin, and different implementations
+  use different methods:
+
+    <!-- Taker creates tx template, see discussion between harding and
+    waxwing: http://gnusha.org/joinmarket/2020-06-14.log -->
+
+    - [Joinmarket][] has the user paying for the coinjoin (the *market
```
Comment (jnewbery) [inline _posts/en/newsletters/2020-06-17-newsletter.md:59]: I think the sentence structure would be simpler, and easier to parse, as "In Joinmarket, the user paying for the coinjoin (the market taker) contacts ...., collects, ... and creates ...

With "(the market taker)" between "has the user paying for the coinjoin" and "contact each..." its a little difficult to parse that 'the user paying' is the subject of a clause.

**Comment by jnewbery** [comment]
ACK the "Changes to services and client software" section. Great stuff, @bitschmidty !

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+          - "Ability to spend directly, or with coinjoin"
+          - "Can specify exact amount of coinjoin (figures from 0.01 to 30.0 btc"
+        -->
+
+        This model gives takers a lot of flexibility with their own
+        inputs and outputs to the transaction template.  For example, a
+        taker can choose the amounts of the coinjoin they want to create
+        or can spend their money to a third party as part of a coinjoin.
+
+    - [Wasabi][] uses a centralized coordinator who organizes every
```
Comment (jonatack) [inline _posts/en/newsletters/2020-06-17-newsletter.md:83]: The coordinator is a "who" in this sentence and an "it" further down in the same paragraph.

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+      coinjoin made using that software.  To prevent that coordinator
+      from learning which inputs fund which outputs, users anonymously
+      commit to the outputs they want to create, receiving a [chaumian
+      blinded signature][] over the commitment.  Later, the user connects
```
Comment (jonatack) [inline _posts/en/newsletters/2020-06-17-newsletter.md:87]: we just described users in the plural, so maybe s/the user/each user/ here

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+        Because the coordinator is unable to view the output at the time
+        it creates its blinded signature, it can't allow the user to
+        specify an arbitrary amount or the user could attempt to receive
+        more money than they contributed to the coinjoin.  This isn't a
+        security risk---the other participants will refuse to sign any
+        malformed transaction---but such a failure requires restarting
+        the protocol.  If arbitrary amounts were allowed, the blinding
+        would prevent identification of the lying user and make it
+        impossible to ban them from future rounds, allowing an unlimited
+        DoS of the protocol.  Instead, Wasabi requires all outputs
```
Comment (jonatack) [inline _posts/en/newsletters/2020-06-17-newsletter.md:104]: maybe s/requires all/requires that all/

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+    anonymous credential for each output commitment that allows the
+    client to later anonymously submit the revealed output to the
+    coordinator for inclusion in the transaction template.
+
+    Software that implements the protocol will be able to create
+    coordinated coinjoins---with privacy from the coordinator---that
+    allow the clients to select their output amounts, facilitating
+    experiments with non-equal value coinjoins (see [Newsletter
+    #79][news79 unequal coinjoin]) and payments made within a coinjoin
+    (either to third-parties not participating in the coinjoin or to
```
Comment (jonatack) [inline _posts/en/newsletters/2020-06-17-newsletter.md:130]: s/third-parties/third parties/

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+    that they collectively sum to a specified value.  The coordinator
+    uses this specified value to verify that the sum of the outputs the
+    client wants to create is equal to the sum of the inputs provided by
+    the client (minus fees).  The coordinator can then provide an
+    anonymous credential for each output commitment that allows the
+    client to later anonymously submit the revealed output to the
+    coordinator for inclusion in the transaction template.
+
+    Software that implements the protocol will be able to create
+    coordinated coinjoins---with privacy from the coordinator---that
```
Comment (jonatack) [inline _posts/en/newsletters/2020-06-17-newsletter.md:126]: "with privacy from the coordinator" seems unclear to me

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+
+## Changes to services and client software
+
+*In this monthly feature, we highlight interesting updates to Bitcoin
+wallets and services.*
+
+- **Caravan adds BIP32, coin selection, hardware wallet test suite:**
+  In addition to single address multisig coordination, Caravan now supports HD
+  wallet multisig coordination and coin selection features. [Unchained Capital
+  also announced][unchained caravan blog] a test suite for testing hardware
```
Comment (jonatack) [inline _posts/en/newsletters/2020-06-17-newsletter.md:147]: maybe only "announced" or "announced a test suite" for the link text

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+title: 'Bitcoin Optech Newsletter #102'
+permalink: /en/newsletters/2020/06/17/
+name: 2020-06-17-newsletter
+slug: 2020-06-17-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter summarizes the CoinPool payment pool proposal and
+the WabiSabi coordinated coinjoin protocol.  Also included are our
```
Comment (jonatack) [inline _posts/en/newsletters/2020-06-17-newsletter.md:11]: Historical note: git grep shows a number of "CoinJoin" spellings in the 2018 newsletters.

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+    with keyed-verification anonymous credentials), so its authors are
+    seeking review, criticism, and suggestions about how the protocol
+    can be used most effectively.
+
+## Changes to services and client software
+
+*In this monthly feature, we highlight interesting updates to Bitcoin
+wallets and services.*
+
+- **Caravan adds BIP32, coin selection, hardware wallet test suite:**
```
Comment (harding) [inline _posts/en/newsletters/2020-06-17-newsletter.md:144]: I think this is coin *control* not coin *selection*.  See: https://twitter.com/murchandamus/status/1256270340008730624

File: _posts/en/newsletters/2020-06-17-newsletter.md [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+- **Bitpay's Copay and Bitcore projects support native segwit:**
+  Bitpay's [Copay wallet][copay segwit] and backend [Bitcore service][bitcore
+  segwit] both now support the creation of and spending from native segwit outputs.
+
+- **Desktop version of Blockstream Green wallet:**
+  Blockstream has released their [Green wallet for desktop][blockstream green desktop blog]
+  on macOS, Windows, and Linux. The desktop version supports 2-of-3 and 2-of-2
+  multisig wallets as well as Tor and testnet support.
+
+- **React native library photon-lib announced:** [Tankred Hase
```
Comment (jonatack) [inline _posts/en/newsletters/2020-06-17-newsletter.md:159]: Perhaps set off "photon-lib" with commas. Here and line 161, I believe the mobile app JS framework is normally written as "React Native".

File: _posts/en/newsletters/2020-06-17-newsletter.md (hunk 10 above)
Comment (harding) [reply _posts/en/newsletters/2020-06-17-newsletter.md:11]: Yeah.  We mostly haven't gone back and applied our style guide rules retroactively.  I don't have a strong preference about whether or not we should do that---I like consistency, but I also consider newsletters older than a week or two to be static documents.

File: _posts/en/newsletters/2020-06-17-newsletter.md (hunk 2 above)
Comment (jonatack) [reply _posts/en/newsletters/2020-06-17-newsletter.md]: When I click on it with a non-recent Firefox (68.9) on Debian/PureOS, I get the usual "What should Firefox do with this file?" dialog asking if I'd like to open the pdf or download it. In my case, I think it depends on the browser settings.

File: _posts/en/newsletters/2020-06-17-newsletter.md (hunk 11 above)
Comment (jonatack) [reply _posts/en/newsletters/2020-06-17-newsletter.md]: Nice catch!

File: _posts/en/newsletters/2020-06-17-newsletter.md (hunk 2 above)
Comment (harding) [reply _posts/en/newsletters/2020-06-17-newsletter.md]: I get the same (also on Firefox), but I tested Chromium and it automatically downloaded to the file system.  By comparison, if I open https://bitcoincore.org/bitcoin.pdf in Chromium, it opens in Chromium's version of pdf.js, which I think is what @adamjonas would prefer happen.

**Comment by jnewbery** [comment]
ACK 9bf7afb8629d83eb6f11473e4030036653821a16

**Comment by bitschmidty** [comment]
Squashed and merged. Thanks for authoring @harding and for reviews from @jonatack @jnewbery @adamjonas !

**Pull request merged**